pkg/kubelet/dockershim/libdocker/helpers.go

```diff
@@ -40,7 +40,7 @@ func ParseDockerTimestamp(s string) (time.Time, error) {
 func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
 	// The image string follows the grammar specified here
```

Comment:
is this comment still accurate with the change to the parsing method we're using?

Reply:
It is not accurate.

The call stack is `CRI.ImageStatus` -> `InspectImageByRef` -> `matchImageTagOrSHA`.

And `CRI.ImageStatus` is supposed to support all image references, including `RepoTag`, `RepoDigest` and `ID`.

Reply:
Hm, the comment of the code is confusing:
```
// matchImageTagOrSHA checks if the given image specifier is a valid image ref,                                                                                                                             
// and that it matches the given image. It should fail on things like image IDs
// (config digests) and other digest-only references, but succeed on image names
// (`foo`), tag references (`foo:bar`), and manifest digest references
// (`foo@sha256:xyz`).
```

But the code is trying to match the digest part of the image reference with image ID...

We should either repurpose the function to support all references, or call both `InspectImageByRef` and `InspectImageByID` in `ImageStatus`, and fix this function.